x-pack/plugins/observability/public/plugin.ts

```diff
@@ -172,13 +176,37 @@ export class Plugin
     }
 
     this.navigationRegistry.registerSections(
-      of([
-        {
-          label: '',
-          sortKey: 100,
-          entries: [{ label: 'Overview', app: 'observability-overview', path: '/overview' }],
-        },
-      ])
+      from(updater$).pipe(
+        map((value) => {
+          const deepLinks = value(app)?.deepLinks ?? [];
+
+          const overviewLink = {
+            label: i18n.translate('xpack.observability.overviewLinkTitle', {
+              defaultMessage: 'Overview',
+            }),
+            app: observabilityAppId,
+            path: '/overview',
+          };
+
+          // Reformat the visible links to be NavigationEntry objects instead of
+          // AppDeepLink objects.
```

Comment:
Can you explain why this is needed?

Reply:
In a comment in the code or just here? The deep links you register with the app and the links you register for the observability-specific navigation use different keys. I can open an issue to make it so they use the same type and I don't have to do this.